### ISSUE TITLE
Not able to recover from failure caused by elasticsearch read-only

### ISSUE THREAD

**Issue by robertyoung81** [confluentinc/kafka-connect-elasticsearch#387]
I initially ran into this issue due to my disk usage going over 85% usage.  Elasticsearch went into read_only_allow_delete mode.  After I cleaned up disk space and cleared the read_only_allow_delete flag, kafka-connect-elasticsearch is unable to recover without manually restarting the task (the same set of log lines from below will just keep repeating).

Since tasks still show as RUNNING when I query from http://<IP>:8087/connectors/elasticsearch-sink/status, I can't figure out a way to detect the issue via API, so need to look at the logs to know that I need to restart the task.  Is there any way to detect this error through the APIs?

```
Failed to execute batch 2 of 1 records with attempt 1/6, will attempt retry after 85 ms. Failure reason: Bulk request failed: [{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"}]
Failed to execute batch 2 of 1 records with attempt 2/6, will attempt retry after 57 ms. Failure reason: Bulk request failed: [{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"}]
Failed to execute batch 2 of 1 records with attempt 3/6, will attempt retry after 371 ms. Failure reason: Bulk request failed: [{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"}]
Failed to execute batch 2 of 1 records with attempt 4/6, will attempt retry after 734 ms. Failure reason: Bulk request failed: [{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"}]
Failed to execute batch 2 of 1 records with attempt 5/6, will attempt retry after 1388 ms. Failure reason: Bulk request failed: [{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"}]
Failed to execute batch 2 of 1 records after total of 6 attempt(s)
org.apache.kafka.connect.errors.ConnectException: Bulk request failed: [{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"}]
        at io.confluent.connect.elasticsearch.bulk.BulkProcessor$BulkTask.execute(BulkProcessor.java:438)
        at io.confluent.connect.elasticsearch.bulk.BulkProcessor$BulkTask.call(BulkProcessor.java:389)
        at io.confluent.connect.elasticsearch.bulk.BulkProcessor$BulkTask.call(BulkProcessor.java:375)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
        at io.confluent.connect.elasticsearch.bulk.BulkProcessor$BulkProcessorThread.run(BulkProcessor.java:370)
WorkerSinkTask{id=elasticsearch-sink-0} Offset commit failed, rewinding to last committed offsets
org.apache.kafka.connect.errors.ConnectException: Bulk request failed: [{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"}]
        at io.confluent.connect.elasticsearch.bulk.BulkProcessor$BulkTask.execute(BulkProcessor.java:438)
        at io.confluent.connect.elasticsearch.bulk.BulkProcessor$BulkTask.call(BulkProcessor.java:389)
        at io.confluent.connect.elasticsearch.bulk.BulkProcessor$BulkTask.call(BulkProcessor.java:375)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
        at io.confluent.connect.elasticsearch.bulk.BulkProcessor$BulkProcessorThread.run(BulkProcessor.java:370)
[Consumer clientId=connector-consumer-elasticsearch-sink-0, groupId=connect-elasticsearch-sink] Seeking to offset 1813070687 for partition mwe.resource.locate-0
[Consumer clientId=connector-consumer-elasticsearch-sink-0, groupId=connect-elasticsearch-sink] Seeking to offset 0 for partition mwe.resource.interaction-0
[Consumer clientId=connector-consumer-elasticsearch-sink-0, groupId=connect-elasticsearch-sink] Seeking to offset 0 for partition mwe.resource.alert-0
[Consumer clientId=connector-consumer-elasticsearch-sink-0, groupId=connect-elasticsearch-sink] Seeking to offset 1778163907 for partition mwe.resource.zoneChange-0
[Consumer clientId=connector-consumer-elasticsearch-sink-0, groupId=connect-elasticsearch-sink] Seeking to offset 0 for partition mwe.resource.configuration-0
WorkerSinkTask{id=elasticsearch-sink-0} Commit of offsets threw an unexpected exception for sequence number 106: null
org.apache.kafka.connect.errors.ConnectException: Bulk request failed: [{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"}]
        at io.confluent.connect.elasticsearch.bulk.BulkProcessor$BulkTask.execute(BulkProcessor.java:438)
        at io.confluent.connect.elasticsearch.bulk.BulkProcessor$BulkTask.call(BulkProcessor.java:389)
        at io.confluent.connect.elasticsearch.bulk.BulkProcessor$BulkTask.call(BulkProcessor.java:375)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
        at io.confluent.connect.elasticsearch.bulk.BulkProcessor$BulkProcessorThread.run(BulkProcessor.java:370)
Can't convert record from topic/partition/offset mwe.resource.locate/0/1813070687. Error message: Bulk request failed: [{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"}]
WorkerSinkTask{id=elasticsearch-sink-0} Offset commit failed, rewinding to last committed offsets
org.apache.kafka.connect.errors.ConnectException: Bulk request failed: [{"type":"cluster_block_exception","reason":"blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];"}]
        at io.confluent.connect.elasticsearch.bulk.BulkProcessor$BulkTask.execute(BulkProcessor.java:438)
        at io.confluent.connect.elasticsearch.bulk.BulkProcessor$BulkTask.call(BulkProcessor.java:389)
        at io.confluent.connect.elasticsearch.bulk.BulkProcessor$BulkTask.call(BulkProcessor.java:375)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
        at io.confluent.connect.elasticsearch.bulk.BulkProcessor$BulkProcessorThread.run(BulkProcessor.java:370)
```

**Comment by levzem** [comment]
@robertyoung81 usually a `ConnectException` will kill the task and show it as failed, but unfortunately because the ES connector executes each request on a separate thread, the `ConnectException` is not properly bubbled up to kill the task. This might be something worth looking into to fix.

**Comment by dainiusjocas** [comment]
I also experienced this issue.

**Comment by levzem** [comment]
@dainiusjocas @dainiusjocas #411 should address this issue

**Comment by robertyoung81** [comment]
@levzem Does this fix only apply to 5.0.x, or has it been merged to 5.3/5.4/5.5 branches as well?

**Comment by levzem** [comment]
@robertyoung81 should be present in all branches including `5.0.x` and later
